lib/src/r_tree/leaf_node.dart

```diff
@@ -28,8 +28,10 @@ class LeafNode<E> extends Node<E> {
     return new LeafNode<E>(branchFactor);
   }
 
-  Iterable<RTreeDatum<E>> search(Rectangle searchRect) {
-    return _items.where((RTreeDatum<E> item) => item.overlaps(searchRect));
+  Iterable<RTreeDatum<E>> search(
+      Rectangle searchRect, bool Function(E item) shouldInclude) {
+    return _items.where((RTreeDatum<E> item) =>
+        shouldInclude(item.value) && item.overlaps(searchRect));
```

Comment:
Do you think it matters for performance which of these checks is first, for short circuiting out if it is false? Do you think we more likely to hit lots of items that don't overlap with the search rect, or lots of items that don't meet our filter criteria?

Reply:
That's a great question. We'll probably hit more items that do not overlap the searchRect, however, that overlaps method is just a little bit of simply integer comparisons, so it'll be fast. The filter argument could be as bad as the implementation provided by the caller 😄  

I'll switch the order